EGATE-83: Event Adapter Deployment

rgauss opened the pull request:
- Renamed activiti-cloud-events-adapter-service to
activiti-cloud-services-events-adapter to follow Activiti naming
precedence
- Changed packages to org.activiti.cloud.services to follow Activiti
naming precedence
- Changed Activiti version to 7.0.0-SNAPSHOT
- Updated events to reference 7.0.0-SNAPSHOT changes
- Added EnableActivitiEventAdapter class

rgauss committed:
EGATE-83: Event Adapter Deployment

- Renamed activiti-cloud-events-adapter-service to
activiti-cloud-services-events-adapter to follow Activiti naming
precedence
- Changed packages to org.activiti.cloud.services to follow Activiti
naming precedence
- Changed Activiti version to 7.0.0-SNAPSHOT
- Updated events to reference 7.0.0-SNAPSHOT changes
- Added EnableActivitiEventAdapter class

rgauss committed:
EGATE-83

- Added spring-cloud-starter-stream-rabbit to the starter

jamalkm commented on activiti-cloud-starter-events-adapter/src/main/java/org/activiti/cloud/starter/events/configuration/EnableActivitiEventAdapter.java:
```diff
@@ -0,0 +1,34 @@
+@Retention(RetentionPolicy.RUNTIME)
+@EntityScan(basePackages = {"org.activiti.cloud.services.events.adapter"})
+@Inherited
+@EnableAutoConfiguration
```
@rgauss it make sense to add these two annotations as well:
```
@EnableDiscoveryClient
@EnableIntegration
```
to eliminate the need of adding them manually when someone adds \@SpringBootApplication

rgauss replied:
Yes, good idea, done.

jamalkm commented on activiti-cloud-starter-events-adapter/src/test/java/org/activiti/cloud/events/adapter/ActivitiCloudEventsAdapterService.java:
```diff
@@ -18,9 +18,11 @@
 
 @SpringBootApplication
+@ComponentScan(basePackages = {"org.activiti.cloud.services","org.activiti.cloud.alfresco","org.springframework.integration.jms"})
 @EnableDiscoveryClient
```
The last two annotations can be replaced with your new "**EnableActivitiEventAdapter**" annotation.

rgauss replied:
That would create a circular dependency.  `EnableActivitiEventAdapter` is in the adapter project which consumes the services project.

jamalkm replied:
I doubt it, as both `EnableActivitiEventAdapter` and the tests are in **activiti-cloud-starter-events-adapter** project.

rgauss replied:
Yes, you're right, I previously moved all the tests to the service rather than the adapter (which we still may want to do in a subsequent increment).

rgauss committed:
EGATE-83: Activiti Events Adapter Deployment (from @jamalkm)

- Added implementation artifacts to pom
- Added IntegrationComponentScan EnableActivitiEventAdapter
- Changed ActivitiCloudEventsAdapterService to use
EnableActivitiEventAdapter

rgauss committed:
EGATE-83: Activiti Event Adapter Deployment

- Added EnableDiscoveryClient and EnableIntegration

rgauss committed:
EGATE-83: Activiti Events Adapter Deployment

- Moved application.properties to test
- Changed spring-cloud-starter-stream-rabbit scope to test
- Added README

rgauss commented:
I think this is ready for final review.

jamalkm reviewed (approved):
LGTM.

salaboy commented on activiti-cloud-services-events-adapter/pom.xml:
```diff
@@ -0,0 +1,158 @@
+      <artifactId>activemq-client</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.reactivestreams</groupId>
```
@cristina-sirbu @rgauss do we need this dep?

rgauss replied:
Yes, the `JmsConfig` class doesn't compile without it.

salaboy commented on activiti-cloud-services-events-adapter/pom.xml:
```diff
@@ -0,0 +1,158 @@
+          <artifactId>activiti-cloud-api-process-model-impl</artifactId>
+      </dependency>
+    <dependency>
+      <groupId>org.springframework.cloud</groupId>
```
@rgauss @cristina-sirbu why do we need this dep for testing?

rgauss replied:
Good catch.  At one point I had move the tests to this module, these were left over from that iteration.

Deleted.

salaboy commented on activiti-cloud-starter-events-adapter/src/test/java/org/activiti/cloud/events/adapter/ActivitiCloudEventsAdapterService.java:
```diff
@@ -15,14 +15,14 @@
-@EnableIntegration
+@EnableActivitiEventAdapter
+@ComponentScan({"org.activiti.cloud","org.activiti.cloud.events","org.springframework.integration.jms"})
 public class ActivitiCloudEventsAdapterService {
```
@rgauss @cristina-sirbu we usually tend to call this class Application or maybe EventAdapterApplication, just to follow what other people does.

rgauss replied:
Renamed to `Application`.

salaboy reviewed (approved):
Besides some very small comments, this is ready to merge.

rgauss committed:
EGATE-83: Activiti Events Adapter Deployment

- Removed unnecessary test dependencies

rgauss committed:
EGATE-83: Activiti Events Adapter Deployment

- Renamed ActivitiCloudEventsAdapterService to Application

rgauss commented:
@salaboy, I believe all comments have been address.

Who typically does the actual merge?

salaboy commented:
@rgauss I will merge now
